### PR TITLE
Mark visualizations as vizjson2

### PR DESCRIPTION
**Pull request by javitonino**

### CODE REVIEW

**Comment by javitonino** [comment]
Please CR @gfiorav 

cc @luisbosque

File: app/controllers/carto/api/visualizations_controller.rb [hunk 1]
```diff
@@ -204,6 +205,13 @@ def to_hash(visualization)
         VisualizationPresenter.new(visualization, current_viewer, self).to_poro
       end
 
+      def mark_as_vizjson2
+        referer_host = URI.parse(request.referer).host
+        unless referer_host.ends_with?('carto.com') || referer_host.ends_with?('cartodb.com')
+          @visualization.mark_as_vizjson2
+        end
+      rescue
```
Comment (gfiorav) [inline app/controllers/carto/api/visualizations_controller.rb]: I'd capture the specific exception. Better not shadow any other errors!

File: app/models/carto/visualization.rb [hunk 2]
```diff
@@ -431,6 +433,14 @@ def state
     super ? super : build_state
   end
 
+  def mark_as_vizjson2
+    $tables_metadata.SADD(V2_VISUALIZATIONS_REDIS_KEY, id)
```
Comment (gfiorav) [inline app/models/carto/visualization.rb]: Id use `SETNX`.

File: app/models/carto/visualization.rb [hunk 3]
```diff
@@ -431,6 +433,14 @@ def state
...
+  end
+
+  def uses_vizjson2?
+    $tables_metadata.SISMEMBER(V2_VISUALIZATIONS_REDIS_KEY, id) > 0
```
Comment (gfiorav) [inline app/models/carto/visualization.rb]: Why not `GET`? Also, something tells we should avoid using this method massively. We don't have the use case now, but we might in the future. Maybe adding a comment?

File: spec/requests/carto/api/visualizations_controller_spec.rb [hunk 4]
```diff
@@ -32,6 +32,17 @@ def vizjson_vx_version
       def attributions_from_vizjson(visualization)
         visualization['layers'][1]['options']['attribution'].split(',').map(&:strip)
       end
+
+      it 'marks visualizations as using vizjson2' do
```
Comment (gfiorav) [inline spec/requests/carto/api/visualizations_controller_spec.rb]: good test!

File: app/models/carto/visualization.rb (hunk 2 above)
Comment (javitonino) [inline app/models/carto/visualization.rb]: I was considering using that (id in key) but finally used sets. Some discussion in https://github.com/CartoDB/cartodb-platform/issues/2506

Comment (gfiorav) [inline app/models/carto/visualization.rb]: Ok! Then disregard the first part of the next comment! 😄

File: app/controllers/carto/api/visualizations_controller.rb (hunk 1 above)
Comment (javitonino) [inline app/controllers/carto/api/visualizations_controller.rb]: In fact, I just realised the rescue should mark the visualization :(

Adding tests for that as first step

**Comment by javitonino** [comment]
Made some changes and added some tests. Can you take a second look?

**Pull request merged**